### PR TITLE
add tomcrypt package

### PR DESCRIPTION
**Pull request by wheybags**
As usual, would like a hunter-packages repo

### CODE REVIEW

**Comment by ruslo** [comment]
Done: https://github.com/hunter-packages/libtomcrypt

**Comment by wheybags** [comment]
ok, changed and force-pushed

File: cmake/projects/tomcrypt/hunter.cmake [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+include(hunter_download)
+include(hunter_pick_scheme)
+include(hunter_cacheable)
+
+# List of versions here...
+hunter_add_version(
+    PACKAGE_NAME
+    tomcrypt
+    VERSION
+    "1.0-p1"
```
Comment (ruslo) [inline cmake/projects/tomcrypt/hunter.cmake]: Should be `1.17-p1`.

**Comment by wheybags** [comment]
whoops, fixed

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1418
* https://travis-ci.org/ingenue/hunter/builds/210048790

File: cmake/projects/tomcrypt/hunter.cmake [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+# List of versions here...
+hunter_add_version(
+    PACKAGE_NAME
+    tomcrypt
+    VERSION
+    "1.17-p1"
+    URL
+    "https://github.com/hunter-packages/libtomcrypt/archive/1.17-p1.tar.gz"
+    SHA1
+    9f088bb9ffcb337f0561c5888c6b36311c19433c
```
Comment (ruslo) [inline cmake/projects/tomcrypt/hunter.cmake]: All tests failed, this one should be recalculated.

File: examples/tomcrypt/main.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+    hash_descriptor[idx].done(&md, hash);
+
+   /* dump to screen */
+   for (x = 0; x < hash_descriptor[idx].hashsize; x++)
+     printf("%02x ", hash[x]);
+
+   printf("\n");
+   return 0;
+}
```
Comment (ruslo) [inline examples/tomcrypt/main.c]: http://0.readthedocs.io/en/latest/git.html#check-there-is-a-new-line-at-end-of-file

**Comment by wheybags** [comment]
Had wrong hash, fixed

File: examples/tomcrypt/main.c (hunk 3 above)
Comment (wheybags) [inline examples/tomcrypt/main.c]: So do you want it to have an ending newline or not? Does it matter really?

File: cmake/projects/tomcrypt/hunter.cmake (hunk 2 above)
Comment (wheybags) [inline cmake/projects/tomcrypt/hunter.cmake]: It's updated now

File: examples/tomcrypt/main.c (hunk 3 above)
Comment (ruslo) [inline examples/tomcrypt/main.c]: > So do you want it to have an ending newline or not?

The section name is `Check there is a new line at end of file`, so...

[Updated](http://0.readthedocs.io/en/latest/git.html#check-there-is-a-new-line-at-end-of-file) anyway:
```
Please fix this warning by adding new line at end of file.
```

> Does it matter really?

Kind of style checking, so it doesn't affect functionality but it's nice to keep things consistent.

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1419
* https://travis-ci.org/ingenue/hunter/builds/210069308

**Comment by ruslo** [comment]
> Testing

Analyze toolchain failed because of warning in code. iOS failed because of log size, I've add a code to set verbosity level low. New testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1420
* https://travis-ci.org/ingenue/hunter/builds/210075293

**Comment by ruslo** [comment]
Everything looks good except `analyze`:
```
/home/travis/build/ingenue/hunter/_testing/Hunter/_Base/1ed109e/33fcb0f/f56fa26/Build/tomcrypt/Source/src/ciphers/aes/aes.c:149:9: warning: Value stored to 'j' is never read

        j = 44;

        ^   ~~

/home/travis/build/ingenue/hunter/_testing/Hunter/_Base/1ed109e/33fcb0f/f56fa26/Build/tomcrypt/Source/src/ciphers/aes/aes.c:162:9: warning: Value stored to 'j' is never read

        j = 52;

        ^   ~~

/home/travis/build/ingenue/hunter/_testing/Hunter/_Base/1ed109e/33fcb0f/f56fa26/Build/tomcrypt/Source/src/ciphers/aes/aes.c:183:9: warning: Value stored to 'j' is never read

        j = 60;

        ^   ~~
```

@wheybags feel free to apply fixes or exclude toolchain.

**Comment by wheybags** [comment]
https://github.com/ingenue/hunter/pull/73 disabled, sorry for the delay

**Comment by ruslo** [comment]
Testing:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1724
* https://travis-ci.org/ingenue/hunter/builds/233107988?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Uploading:
* https://ci.appveyor.com/project/ingenue/hunter/build/1.0.1726
* https://travis-ci.org/ingenue/hunter/builds/233114202?utm_source=github_status&utm_medium=notification

**Comment by ruslo** [comment]
Released: https://github.com/ruslo/hunter/releases/tag/v0.18.60

Feel free to update wiki/docs.

**Pull request merged**